src/diffusers/pipelines/hunyuandit/pipeline_hunyuandit.py

```diff
@@ -0,0 +1,1010 @@
+                               num_images_per_prompt,
+                               do_classifier_free_guidance,
+                               negative_prompt,
+                               prompt_embeds=prompt_embeds_t5,
+                               negative_prompt_embeds=negative_prompt_embeds_t5,
+                               lora_scale=text_encoder_lora_scale,
+                               embedder={'model': self.embedder_t5, 'tokenizer': self.tokenizer_t5, 'max_length': 256},
+                               )
+
+        # For classifier free guidance, we need to do two forward passes.
+        # Here we concatenate the unconditional and text embeddings into a single batch
+        # to avoid doing two forward passes
+        if do_classifier_free_guidance:
+            prompt_embeds = torch.cat([negative_prompt_embeds, prompt_embeds])
+            attention_mask = torch.cat([uncond_attention_mask, attention_mask])
+            prompt_embeds_t5 = torch.cat([negative_prompt_embeds_t5, prompt_embeds_t5])
+            attention_mask_t5 = torch.cat([uncond_attention_mask_t5, attention_mask_t5])
+        
+        prompt_embeds = prompt_embeds.to(dtype=self.transformer.dtype)
+        attention_mask = attention_mask.to(dtype=self.transformer.dtype)
+        prompt_embeds_t5 = prompt_embeds_t5.to(dtype=self.transformer.dtype)
+        attention_mask_t5 = attention_mask_t5.to(dtype=self.transformer.dtype)
+
+        # 5. Prepare timesteps
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps = self.scheduler.timesteps
+
+        # 6. Prepare latent variables
+        num_channels_latents = self.transformer.config.in_channels
+        latents = self.prepare_latents(batch_size * num_images_per_prompt,
+                                       num_channels_latents,
+                                       height,
+                                       width,
+                                       prompt_embeds.dtype,
+                                       device,
+                                       generator,
+                                       latents,
+                                       )
+
+        # 7. Prepare extra step kwargs. TODO: Logic should ideally just be moved out of the pipeline
+        extra_step_kwargs = self.prepare_extra_step_kwargs(generator, eta)
+        
+        # ========================================================================
+        # Arguments: style. (A fixed argument. Don't Change it.)
+        # ========================================================================
+        style = torch.as_tensor([0, 0] * batch_size, device=self._execution_device)
+
+        # ========================================================================
+        # Inner arguments: image_meta_size (Please refer to SDXL.)
+        # ========================================================================
+        size_cond = [1024, 1024] + [width, height, 0, 0]
```

Comment:
idealy we do the same thing as SDXL with this method 
https://github.com/huggingface/diffusers/blob/0ab63ff6478b7cc6b5ae0d46c7c386d476cfa87f/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L725

1. we have pipeline argument `original_size` and `target_size`
2. default them to (`height`, `width`) https://github.com/huggingface/diffusers/blob/0ab63ff6478b7cc6b5ae0d46c7c386d476cfa87f/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L1039
3.  we should probably rename `image_meta_size` to`add_time_ids` 
                   
                     add_time_ids = self._get_add_time_ids(...)

Reply:
I'm kind of confused what we should do here...yiyi could you make an example? thanks